app/js/lib/server/index.js

```diff
@@ -100,6 +100,12 @@ export default class Server extends EventDispatcher {
    * @return {Promise}
    */
   subscribeToNotifications(resubscribe = false) {
+    if (!this.isLoggedIn) {
+      return Promise.reject(new Error(
+        'Subscribing to push notifications is not possible ' +
+        'because the user is not logged in yet.'
+      ));
```

Comment:
note: initially I made this silent because I didn't want an error to be displayed at initial startup. up to you :)